### PR TITLE
Pass accuracy param through erf and atan2 primitives

### PR DESCRIPTION
**Pull request by v0i0**
erf_p.bind() and atan2_p.bind() do not pass the accuracy parameter,
but their Pallas Mosaic lowering rules require it. This causes a
TypeError when using jax.lax.erf in a Pallas TPU kernel.

Match the pattern used by exp and other math primitives (see lax.py
exp_p.bind(x, accuracy=accuracy)).

Authored with assistance from Claude

### CODE REVIEW

File: jax/_src/lax/lax.py [hunk 1]
```diff
@@ -719,7 +719,7 @@ def cos(x: ArrayLike, accuracy=None) -> Array:
   return cos_p.bind(x, accuracy=accuracy)
 
 @export
-def atan2(x: ArrayLike, y: ArrayLike) -> Array:
+def atan2(x: ArrayLike, y: ArrayLike, accuracy=None) -> Array:
```
Comment (jakevdp) [inline jax/_src/lax/lax.py]: Please make `accuracy` keyword-only, and add a type annotation. Also, since this is a public API, we should update the docstring to describe the new parameter.

File: jax/_src/lax/special.py [hunk 2]
```diff
@@ -113,9 +113,9 @@ def bessel_i1e(x: ArrayLike) -> Array:
   """
   return bessel_i1e_p.bind(x)
 
-def erf(x: ArrayLike) -> Array:
+def erf(x: ArrayLike, accuracy=None) -> Array:
```
Comment (jakevdp) [inline jax/_src/lax/special.py]: Same comments here.

**Review by jakevdp — CHANGES_REQUESTED** [review]
Thanks!

A couple comments below.

**Comment by jakevdp** [comment]
Another option here would be for the pallas/mlir lowering rules to provide a default value for the `accuracy` parameter – that would depend on whether it's something we expect users to specify when using the `jax.lax` APIs.

cc/ @apaszke

**Comment by v0i0** [comment]
> Another option here would be for the pallas/mlir lowering rules to provide a default value for the `accuracy` parameter – that would depend on whether it's something we expect users to specify when using the `jax.lax` APIs.
> 
> cc/ @apaszke

i figured i'd follow the pattern of the other special functions, but happy to change it if that makes more sense!

**Comment by jakevdp** [comment]
> i figured i'd follow the pattern of the other special functions, but happy to change it if that makes more sense!

Thanks - yeah I think threading `accuracy` arguments through `jax.lax` is the correct approach here after all. Please refer to the comments regarding docs, annotations, and keyword-only arguments (I'll follow-up to adjust the rest of the file in the same way). You can copy the docstring from one of the other functions where `accuracy` is documented.

**Comment by jakevdp** [comment]
You can follow the precedent used in #35818.

**Comment by v0i0** [comment]
> You can follow the precedent used in https://github.com/jax-ml/jax/pull/35818.

have a look at the change now, tried to follow it like `exp`

**Review by jakevdp — COMMENTED** [review]
Looks good, thanks!

One uncertainty I have here is that although the `pallas/mosaic` lowering rules support the `accuracy` param, the `pallas/mosaic_gpu` lowering rules do not: 

https://github.com/jax-ml/jax/blob/6466a62a07e206e13ff52ff49e730bc2525f7a24/jax/_src/pallas/mosaic_gpu/lowering.py#L2751-L2754

https://github.com/jax-ml/jax/blob/6466a62a07e206e13ff52ff49e730bc2525f7a24/jax/_src/pallas/mosaic_gpu/lowering.py#L2739-L2742

I suspect this may cause problems – maybe @apaszke knows the best way forward here.

**Comment by v0i0** [comment]
alright i tried to make CI / mosaic_gpu happy as well
